models/stored/tx_body.go

```diff
@@ -0,0 +1,141 @@
+package stored
+
+import (
+	"encoding/binary"
+
+	"github.com/Worldcoin/hubble-commander/models"
+	"github.com/Worldcoin/hubble-commander/models/enums/txtype"
+)
+
+const (
+	txTransferBodyLength        = 4
+	txCreate2TransferBodyLength = models.PublicKeyLength + 5
+	txMassMigrationBodyLength   = 4
+)
+
+func txBody(data []byte, transactionType txtype.TransactionType) (TxBody, error) {
+	switch transactionType {
+	case txtype.Transfer:
+		body := new(TxTransferBody)
+		err := body.SetBytes(data)
+		return body, err
+	case txtype.Create2Transfer:
+		body := new(TxCreate2TransferBody)
+		err := body.SetBytes(data)
+		return body, err
+	case txtype.MassMigration:
+		body := new(TxMassMigrationBody)
+		err := body.SetBytes(data)
+		return body, err
+	}
+	return nil, nil
+}
+
+type TxBody interface {
+	ByteEncoder
+	BytesLen() int
+	ToGenericTransaction(base *models.TransactionBase) models.GenericTransaction
+}
+
+func NewTxBody(tx models.GenericTransaction) TxBody {
+	switch tx.Type() {
+	case txtype.Transfer:
+		return &TxTransferBody{
+			ToStateID: tx.ToTransfer().ToStateID,
+		}
+	case txtype.Create2Transfer:
+		return &TxCreate2TransferBody{
+			ToPublicKey: tx.ToCreate2Transfer().ToPublicKey,
+			ToStateID:   tx.ToCreate2Transfer().ToStateID,
+		}
+	case txtype.MassMigration:
+		return &TxMassMigrationBody{
+			SpokeID: tx.ToMassMigration().SpokeID,
+		}
+	}
+	panic("unknown transaction type")
+}
+
+type TxTransferBody struct {
+	ToStateID uint32
+}
+
+func (t *TxTransferBody) Bytes() []byte {
+	b := make([]byte, txTransferBodyLength)
+	binary.BigEndian.PutUint32(b, t.ToStateID)
+	return b
+}
+
+func (t *TxTransferBody) SetBytes(data []byte) error {
+	t.ToStateID = binary.BigEndian.Uint32(data)
+	return nil
+}
+
+func (t *TxTransferBody) BytesLen() int {
+	return txTransferBodyLength
+}
+
+func (t *TxTransferBody) ToGenericTransaction(base *models.TransactionBase) models.GenericTransaction {
+	return &models.Transfer{
+		TransactionBase: *base,
+		ToStateID:       t.ToStateID,
+	}
+}
+
+type TxCreate2TransferBody struct {
+	ToPublicKey models.PublicKey
+	ToStateID   *uint32
```

Comment:
I think we talked about doing this in a future PR, agree that this is a worthwhile change(!) but it's a big enough refactor that I don't want it to delay this PR

Reply:
After a longer thought, this would require splitting `models.Create2Transfer` into two types as well, something like:  `models.PendingCreate2Transfer` and `models.BatchedCreate2Transfer`. Not sure if this is worth the effort right now.